File: SpringBoot/src/main/java/com/sharkfitness/service/AuthService.java
package com.sharkfitness.service;

import com.sharkfitness.dto.LoginRequest;
import com.sharkfitness.dto.RegisterRequest;
import com.sharkfitness.vo.LoginResponse;

/**
 * Service interface for authentication operations
 */
public interface AuthService {
    
    /**
     * Register a new user
     * @param request registration request containing username, password, and role
     * @return LoginResponse with token and user information
     */
    LoginResponse register(RegisterRequest request);
    
    /**
     * Authenticate user and generate token
     * @param request login request containing username and password
     * @return LoginResponse with token and user information
     */
    LoginResponse login(LoginRequest request);
}

File: VUE/src/services/authService.js
import api from './api'

export const authService = {
  login(username, password) {
    return api.post('/api/auth/login', { username, password })
  },

  register(userData) {
    return api.post('/api/auth/register', userData)
  },

  logout() {
    return api.post('/api/auth/logout')
  },

  getCurrentUser() {
    return api.get('/api/auth/current-user')
  }
}
